cli/cmd/metrics_diagnostics_util.go

```diff
@@ -128,12 +128,15 @@ func getMetrics(
 		}(pod)
 	}
 
+	timeout := time.NewTimer(waitingTime)
+	defer timeout.Stop()
 wait:
 	for {
+		timeout.Reset(waitingTime)
```

Comment:
The docs state that "For a Timer created with NewTimer, Reset should be invoked only on stopped or expired timers with drained channels". Not sure if this applies in this non-concurrent case, but this might get flagged just as the `time.After` usage was.

Reply:
After looking at this some more, I think the intention was for this timeout to apply to getting all of the results, rather than being a timeout for each individual result.  Therefore, I just removed the Reset altogether.